### PR TITLE
GH-125521: Remove `if (true)` from generated output to reduce C compiler warnings

### PR DESCRIPTION
**Pull request by markshannon**
* Issue: gh-125521

### CODE REVIEW

File: Tools/cases_generator/generators_common.py [hunk 1]
```diff
@@ -165,16 +165,24 @@ def error_if(
         storage: Storage,
         inst: Instruction | None,
     ) -> bool:
-        self.out.emit_at("if ", tkn)
         lparen = next(tkn_iter)
-        self.emit(lparen)
         assert lparen.kind == "LPAREN"
         first_tkn = tkn_iter.peek()
-        emit_to(self.out, tkn_iter, "COMMA")
```
Comment (iritkatriel) [inline Tools/cases_generator/generators_common.py:173]: Wouldn't it be simpler to just add an unconditional `ERROR()` macro to the DSL?

Comment (markshannon) [reply Tools/cases_generator/generators_common.py:173]: That wouldn't prevent anyone writing `ERROR_IF(true, ...)` so we'd still need to handle it.

**Review by iritkatriel — APPROVED** [review]

**Comment by iritkatriel** [comment]
Seems like there is no test for this.

File: Tools/cases_generator/analyzer.py [hunk 2]
```diff
@@ -744,7 +744,7 @@ def always_exits(op: parser.InstDef) -> bool:
             if tkn.text == "DEOPT_IF" or tkn.text == "ERROR_IF":
                 next(tkn_iter)  # '('
                 t = next(tkn_iter)
-                if t.text == "true":
+                if t.text in ("true", "1"):
```
Comment (iritkatriel) [inline Tools/cases_generator/analyzer.py:747]: I think you could do `eval(f"bool({t.text})")`

Comment (iritkatriel) [reply Tools/cases_generator/analyzer.py:747]: Actually no, it's `true` rather than `True`.

**Review by iritkatriel — APPROVED** [review]

**Pull request merged**